scripts/Tools/case.setup

```diff
@@ -45,9 +49,16 @@ OR
     parser.add_argument("-r", "--reset", action="store_true",
                         help="Does a clean followed by setup")
 
+    parser.add_argument("-a", "--apply-usermods",
+                        help="Specify a directory to apply user mods")
+
     args = CIME.utils.parse_args_and_handle_standard_logging_options(args, parser)
 
-    return args.caseroot, args.clean, args.test_mode, args.reset
+    if args.apply_usermods is not None:
```

Comment:
You might consider using the mutually exclusive options capability of parser here, as an example xmlquery uses this.